clang-tools-extra/clangd/TidyProvider.cpp

```diff
@@ -219,6 +219,9 @@ TidyProvider disableUnusableChecks(llvm::ArrayRef<std::string> ExtraBadChecks) {
       "-bugprone-use-after-move",
       // Alias for bugprone-use-after-move.
       "-hicpp-invalid-access-moved",
+      // Check uses dataflow analysis, which might hang/crash unexpectedly on
+      // incomplete code.
+      "-bugprone-unchecked-optional-access",
```

Comment:
A consequence of putting the check here, is that a user who wants the check in spite of its bugginess does not have a way to enable it.

It was suggested, in https://github.com/clangd/clangd/issues/1700#issuecomment-1659690492 and https://github.com/clangd/clangd/issues/1337#issuecomment-1610876189, that we should have a way to allow users to override the disablements in this "bad checks" list. Could you weigh in on that proposal?

Reply:
FWIW, we had this check enabled internally before it was upstream'd and experience on incomplete code has always been problematic. Hence I don't see much point in letting people run this on an incomplete AST. I believe the recommended workflow should be through running clang-tidy binary directly. I'll add some comments on those threads too.